portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -239,8 +241,21 @@ public DLFileEntry addFileEntry(
 			PortalUtil.getCurrentAndAncestorSiteGroupIds(groupId), folderId,
 			fileEntryTypeId);
 
+		String actualExtension = null;
+
+		if (inputStream instanceof ByteArrayFileInputStream) {
+			ByteArrayFileInputStream inputByteArrayFileInputStream =
+				(ByteArrayFileInputStream)inputStream;
+
+			String actualName = inputByteArrayFileInputStream.getFile(
+			).getName();
+
+			actualExtension = FileUtil.getExtension(actualName);
```

Comment:
Please, avoid chaining ...
```
File file =  inputByteArrayFileInputStream.getFile();

actualExtension = FileUtil.getExtension(file.getName());
```

Reply:
Ok, I misunderstood you.

Actually, what you proposed is what I was trying to avoid. I don't like instancing the whole Object File to access just a substring.

Anyway I have to admit that the resulting code is much cleaner.

Reply:
Changed